### DISCUSSION TITLE
add filtered bookmarks from api.. is this possible?

### DISCUSSION THREAD

**Discussion by eliteproxy7** [directus/directus#6714] [Archive]
I want to add a Bookmark preset with a filtered view, how can I do this from the API? Looks like I have to add the filter first and then add the bookmark and the filter has a unique key value. 

can all this be done from the api?

**Comment by rijkvanzanten** [comment]
> can all this be done from the api?

The Directus app uses the API for everything as well, so whatever can be done in the app can be done through the API as well 👍🏻 

The `/presets` endpoint itself is fairly dumb. If you save the structure it returns, you'll create a new bookmark. The structure of filters is a little different than expected, but you should be able to copy and alter that from an existing one. The "key" under the filter objects is just a random string the app uses to reliably keep track of what filters are applied and shown, and can be set to anything (as long as they're unique)

**Comment by jjgonc** [comment]
Hey, can you tell me how did you fully access the API? Or is it just the directory you created while creating the project?
I'm new in this "world" and I'm trying to get access to customise endpoints authentication and things like that.
What do I have to do and how can I do it?
Thank's a lot!

**Reply by philleepflorence** [reply]
Hi, hope this helps answer your question regarding filtered bookmarks and the API:

- [Introduction - API Reference](https://docs.directus.io/reference/api/introduction/)
- [Authentication - API Reference](https://docs.directus.io/reference/api/authentication/)
- [Managing Presets - API Reference](https://docs.directus.io/reference/api/system/presets/)
- [Custom API Endpoints - Guide](https://docs.directus.io/guides/api-endpoints/)
